pilot/pkg/proxy/envoy/discovery.go

```diff
@@ -480,6 +484,9 @@ func (ds *DiscoveryService) ConfigUpdate(full bool) {
 	defer clearCacheMutex.Unlock()
 
 	if full {
+		if !ds.fullPush && debugDebounce {
```

Comment:
this is very confusing and unintuitive. I agree with @incfly that the naming/structure of this stuff is not clear. Can you please rename these booleans?

Reply:
Proposals ? Hard to find better names - those 2 reflect what I think the code is intending to do.

Reply:
This becomes
```
if wantFullPush {
  if !ds.doFullPushNext ..
```

A more easier way to understand the code would be when you use enums.. such that
```
ConfigUpdate(pushModeRequested PushMode) {
 if pushModeRequested==PushFull {
   if ds.plannedPushMode == PushIncremental && debugDebounce
```

That said, all this is for just the logging..